scripts/github-release.js

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env node
+
+/** Copyright (c) 2017 Uber Technologies, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// @flow
+/* eslint-env node */
+
+const fetch = require('isomorphic-fetch');
+
+if (!process.env.GITHUB_AUTH_TOKEN) {
+  throw new Error('No GITHUB_AUTH_TOKEN set.');
+}
+
+const repo = 'uber/baseweb';
+const root = `https://api.github.com/repos/${repo}`;
+const headers = {Authorization: `token ${process.env.GITHUB_AUTH_TOKEN}`};
+
+// https://github.com/uber-workflow/probot-app-release-notes/blob/master/pr-for-commit.js
+async function prFromCommit(sha) {
+  const res = await fetch(`https://github.com/${repo}/branch_commits/${sha}`);
+  if (res.ok) {
+    const text = await res.text();
+    const mergedRegex = /<li class="pull-request">\((?:.+?)(\d+)<\/a>\)<\/li>/;
+    const pr = text.match(mergedRegex);
+    if (pr) {
+      return parseInt(pr[1], 10);
+    }
+  } else {
+    throw new Error(res.statusText);
+  }
+}
+
+async function main() {
+  const tagsResponse = await fetch(`${root}/tags`, {headers});
+  if (!tagsResponse.ok) {
+    throw new Error(`Failed to fetch tags. ${tagsResponse.statusText}`);
+  }
+  const tags = await tagsResponse.json();
+
+  const commitsResponse = await fetch(`${root}/commits`, {headers});
+  if (!commitsResponse.ok) {
+    throw new Error(`Failed to fetch commits. ${commitsResponse.statusText}`);
+  }
+  const [releaseCommit, ...commits] = await commitsResponse.json();
+
+  const releaseMessageRgx = /^Release (v[0-9]+\.[0-9]+\.[0-9]+)/;
+  const match = releaseMessageRgx.exec(releaseCommit.commit.message);
+  if (!Array.isArray(match) || typeof match[1] !== 'string') {
+    throw new Error(
+      `Failed to parse release commit message: ${releaseCommit.commit.message}`,
+    );
+  }
+  const releaseVersion = match[1];
+
+  const commitsSinceTag = [];
+  for (let commit of commits) {
+    if (commit.sha === tags[0].commit.sha) {
+      break;
+    }
+    commitsSinceTag.push(commit);
+  }
+  const prs = await Promise.all(commitsSinceTag.map(c => prFromCommit(c.sha)));
+
+  const changelogEntries = [];
```

Comment:
do I understand correctly, that this would not group commits based on the label the PR was assigned? not a big deal, if we want it later, we can do that as a followup

Reply:
Correct, I found it not very helpful to organize since we release often. There tends to be maybe 10 prs at most